picasso/src/test/java/com/squareup/picasso3/NetworkRequestHandlerTest.java

```diff
@@ -119,11 +130,18 @@
     assertThat(networkHandler.shouldRetry(true, info)).isFalse();
   }
 
-  @Test public void noCacheAndKnownContentLengthDispatchToStats() throws Exception {
+  @Test public void noCacheAndKnownContentLengthDispatchToStats() {
     responses.add(responseOf(ResponseBody.create(null, new byte[10])));
     Action action = TestUtils.mockAction(URI_KEY_1, URI_1);
-    networkHandler.load(action.getRequest(), 0);
-    verify(stats).dispatchDownloadFinished(10);
+    networkHandler.load(action.getRequest(), 0, new Callback() {
+      @Override public void onSuccess(Result result) {
+        verify(stats).dispatchDownloadFinished(10);
```

Comment:
have a countdownlatch, set an atomic reference here, latch.countDown here, latch.await(10, SECONDS) on the test's main thread, and then verify the result in the atomic reference on the test's main thread.

otherwise, this test won't fail because the assertions are off the test's main thread.